data/src/main/java/com/globant/equattrocchio/data/ImagesServicesImpl.java

```diff
@@ -1,47 +1,97 @@
 
     private static final String URL= "http://splashbase.co/";
-    private Gson gson = new Gson();
+    SplashbaseApi api;
+    private ResultDataMapper resultDataMapper;
+    private ImageDataMapper imageDataMapper;
 
-    @Override
-    public void getLatestImages(final Observer<String> observer) {
+    public ImagesServicesImpl() {
         Retrofit retrofit = new Retrofit.Builder().
                 baseUrl(URL).
                 addConverterFactory(GsonConverterFactory.create())
+                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                 .build();
+        api  = retrofit.create(SplashbaseApi.class);
+    }
 
-        SplashbaseApi api  = retrofit.create(SplashbaseApi.class);
+    @Override
+    public void getLatestImages(final Observer<List<ImageEntity>> observer) {
+        api.getImages()
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new DefaultObserver<Result>() {
+                    @Override
+                    public void onNext(Result result) {
+                        observer.onNext(getResultDataMapper().transform(result, null));
+                    }
 
-        Call<Result> call = api.getImages();
+                    @Override
+                    public void onError(Throwable e) {
+                        observer.onError(new RuntimeException("getLatestImages service error"));
+                    }
 
-        call.enqueue(new Callback<Result>() {
-            @Override
-            public void onResponse(Call<Result> call, Response<Result> response) {
-                if (response.isSuccessful()) {
-                    observer.onNext(gson.toJson(response.body()));
-                }
-            }
+                    @Override
+                    public void onComplete() {
+                        observer.onComplete();
+                    }
+                });
+    }
 
-            @Override
-            public void onFailure(Call<Result> call, Throwable t) {
-                //todo: update the UI with a connection error message
-            }
-        });
+    @Override
+    public void getImageDetail(int id, final Observer<ImageEntity> observer) {
+        api.getImageDetail(id)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new DefaultObserver<Image>() {
+                    @Override
+                    public void onNext(Image image) {
+                        observer.onNext(getImageDataMapper().transform(image, null));
 
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+                        observer.onError(new RuntimeException("getImageDetail service error"));
+                    }
+
+                    @Override
+                    public void onComplete() {
+                        observer.onComplete();
```

Comment:
Please use SingleObserver which does not have the onComplete as mandatory

Reply:
I use the onComplete for the image detail. For hide the progressBar. And the SingleObserver has the onSubscribed as mandatory.